flytekit/clients/auth/auth_client.py

```diff
@@ -281,7 +282,10 @@ def _request_authorization_code(self):
         query = _urlencode(self._request_auth_code_params)
         endpoint = _urlparse.urlunparse((scheme, netloc, path, None, query, None))
         logging.debug(f"Requesting authorization code through {endpoint}")
-        _webbrowser.open_new_tab(endpoint)
+
+        success = _webbrowser.open_new_tab(endpoint)
+        if not success:
+            click.secho(f"Please open the following link in your browser to authenticate:\n{endpoint}")
```

Comment:
nit: why a new line ?

Reply:
Aesthetic preference, happy to remove it though if you prefer :)

Reply:
it can be confusing IMO, but I won't block the PR on this.

Reply:
https://github.com/flyteorg/flytekit/pull/1947/commits/65b1129e2e3146d0af6909c443e98bf7f0d19543